opaque_keys/edx/locator.py

```diff
@@ -121,8 +121,12 @@ def parse_url(cls, string):
         with key 'id' and optional keys 'branch' and 'version_guid'.
 
         Raises:
-            InvalidKeyError: if string cannot be parsed.
+            InvalidKeyError: if string cannot be parsed -or- string ends with a newline.
         """
+        # URL_RE above cannot detect a single trailing newline, so guard against them.
```

Comment:
The regular expression itself *does* catch trailing spaces. The problem was with the `$` regex operator which, according to the Python `re` docs, does the following: `Matches the end of the string or just before the newline at the end of the string`. The regex was doing the former and parsing the trailing newline just fine.